### ISSUE TITLE
Best [objective] So Far Report on Command Line Stuck at First Trial Score

### ISSUE THREAD

**Issue by brethvoice** [keras-team/keras-tuner#693]
**Describe the bug**
The "best ____ so far" output of KerasTuner's command line interface (CLI) output is wrong.

**To Reproduce**
Use any validation metric as the objective other than validation loss. You will get results similar to the attached output screenshot.
![2022-06-02_Best_Objective_Score_So_Far_Not_Updating](https://user-images.githubusercontent.com/11823408/171703281-08dc0291-6009-4f98-aefe-c0b20d85adb7.PNG)

**Expected behavior**
The "best ____ so far" output is consistent with the other outputs on the CLI interface display. It is clear that the objective metric (validation categorical accuracy in the screen shot) is far above the initial value of 0.82, which was established on the very first trial.

**Additional context**
We are using a custom BestMetricCallback. This simply appends information to TensorBoard logs and may be unrelated to KerasTuner's behavior. However, just in case it might be, I have attached the source code for it:
[BestMetricCallback.zip](https://github.com/keras-team/keras-tuner/files/8826767/BestMetricCallback.zip)

**Would you like to help us fix it?**
No, thank you.

**Comment by brethvoice** [comment]
I did not get a chance to copy the trial.json file before it was overwritten at the end of the next trial where `use_adversarial` was set to `True` but I did verify that it had all of the conditional hyperparameters. And after I refreshed the browser for TensorBoard, the new trials and columns for the conditional hyperparameters started showing up. Actually until I refreshed my browser, TensorBoard was stuck displaying the first trial where the boolean conditional hyperparameter was set to `True` and not showing any trials after it. Just hitting the little circle in TensorBoard to "update" it was ineffective.
